### Eclipse Workspace Patch 1.0
#P org.eclipse.core.databinding
Index: src/org/eclipse/core/databinding/BindingException.java
===================================================================
RCS file: /cvsroot/eclipse/org.eclipse.core.databinding/src/org/eclipse/core/databinding/BindingException.java,v
retrieving revision 1.5
diff -u -r1.5 BindingException.java
--- src/org/eclipse/core/databinding/BindingException.java	21 Mar 2007 15:46:14 -0000	1.5
+++ src/org/eclipse/core/databinding/BindingException.java	20 Oct 2007 01:47:39 -0000
@@ -11,11 +11,11 @@
 package org.eclipse.core.databinding;
 
 import java.io.PrintStream;
-import java.io.PrintWriter;
+//import java.io.PrintWriter;
 
 /**
  * An unchecked exception indicating a binding problem.
- * 
+ *
  * @since 1.0
  */
 public class BindingException extends RuntimeException {
@@ -28,7 +28,7 @@
 
 	/**
 	 * Creates a new BindingException with the given message.
-	 * 
+	 *
 	 * @param message
 	 */
 	public BindingException(String message) {
@@ -37,7 +37,7 @@
 
 	/**
 	 * Creates a new BindingException with the given message and cause.
-	 * 
+	 *
 	 * @param message
 	 * @param cause
 	 */
@@ -54,11 +54,11 @@
 		}
 	}
 
-	public void printStackTrace(PrintWriter err) {
-		super.printStackTrace(err);
-		if (cause != null) {
-			err.println("caused by:"); //$NON-NLS-1$
-			cause.printStackTrace(err);
-		}
-	}
+//	public void printStackTrace(PrintWriter err) {
+//		super.printStackTrace(err);
+//		if (cause != null) {
+//			err.println("caused by:"); //$NON-NLS-1$
+//			cause.printStackTrace(err);
+//		}
+//	}
 }
