### Eclipse Workspace Patch 1.0
#P org.eclipse.core.databinding
Index: src/org/eclipse/core/internal/databinding/IdentityWrapper.java
===================================================================
RCS file: /cvsroot/eclipse/org.eclipse.core.databinding/src/org/eclipse/core/internal/databinding/IdentityWrapper.java,v
retrieving revision 1.1
diff -u -r1.1 IdentityWrapper.java
--- src/org/eclipse/core/internal/databinding/IdentityWrapper.java	10 Nov 2006 17:44:52 -0000	1.1
+++ src/org/eclipse/core/internal/databinding/IdentityWrapper.java	20 Oct 2007 02:11:40 -0000
@@ -12,13 +12,15 @@
 
 package org.eclipse.core.internal.databinding;
 
+import org.eclipse.core.internal.runtime.Util;
+
 /**
  * Used for wrapping objects that define their own implementations of equals()
  * and hashCode() when putting them in sets or hashmaps to ensure identity
  * comparison.
- * 
+ *
  * @since 1.0
- * 
+ *
  */
 public class IdentityWrapper {
 	final Object o;
@@ -29,7 +31,7 @@
 	public IdentityWrapper(Object o) {
 		this.o = o;
 	}
-	
+
 	/**
 	 * @return the unwrapped object
 	 */
@@ -38,7 +40,7 @@
 	}
 
 	public boolean equals(Object obj) {
-		if (obj == null || obj.getClass() != IdentityWrapper.class) {
+		if (obj == null || ! Util.isSameClass(this,obj)) {
 			return false;
 		}
 		return o == ((IdentityWrapper) obj).o;
