### Eclipse Workspace Patch 1.0
#P org.eclipse.core.databinding
Index: src/org/eclipse/core/databinding/observable/list/ListDiff.java
===================================================================
RCS file: /cvsroot/eclipse/org.eclipse.core.databinding/src/org/eclipse/core/databinding/observable/list/ListDiff.java,v
retrieving revision 1.3
diff -u -r1.3 ListDiff.java
--- src/org/eclipse/core/databinding/observable/list/ListDiff.java	16 Mar 2007 18:00:56 -0000	1.3
+++ src/org/eclipse/core/databinding/observable/list/ListDiff.java	20 Oct 2007 01:55:11 -0000
@@ -11,40 +11,42 @@
 
 package org.eclipse.core.databinding.observable.list;
 
+import org.eclipse.core.internal.runtime.Util;
+
 
 /**
  * Object describing a diff between two lists.
- * 
+ *
  * @since 1.0
  */
 public abstract class ListDiff {
 
 	/**
 	 * Returns a list of ListDiffEntry
-	 * 
+	 *
 	 * @return a list of ListDiffEntry
 	 */
 	public abstract ListDiffEntry[] getDifferences();
-	
+
 	/**
 	 * @see java.lang.Object#toString()
 	 */
 	public String toString() {
 		ListDiffEntry[] differences = getDifferences();
 		StringBuffer buffer = new StringBuffer();
-		buffer.append(getClass().getName());
-		
+		buffer.append(Util.getClassNameAsString(this));
+
 		if (differences == null || differences.length == 0) {
 			buffer
 				.append("{}"); //$NON-NLS-1$
 		} else {
 			buffer
 				.append("{"); //$NON-NLS-1$
-			
+
 			for (int i = 0; i < differences.length; i++) {
 				if (i > 0)
 					buffer.append(", "); //$NON-NLS-1$
-				
+
 				buffer
 					.append("difference[") //$NON-NLS-1$
 					.append(i)
@@ -54,7 +56,7 @@
 			}
 			buffer.append("}"); //$NON-NLS-1$
 		}
-		
+
 		return buffer.toString();
 	}
 }
