### Eclipse Workspace Patch 1.0
#P org.eclipse.core.databinding
Index: src/org/eclipse/core/databinding/observable/StalenessObservableValue.java
===================================================================
RCS file: /cvsroot/eclipse/org.eclipse.core.databinding/src/org/eclipse/core/databinding/observable/StalenessObservableValue.java,v
retrieving revision 1.1
diff -u -r1.1 StalenessObservableValue.java
--- src/org/eclipse/core/databinding/observable/StalenessObservableValue.java	22 Nov 2007 21:56:15 -0000	1.1
+++ src/org/eclipse/core/databinding/observable/StalenessObservableValue.java	4 Dec 2007 16:18:31 -0000
@@ -14,7 +14,7 @@
 
 /**
  * @since 1.1
- * 
+ *
  */
 class StalenessObservableValue extends AbstractObservableValue {
 
@@ -52,7 +52,7 @@
 	}
 
 	public Object getValueType() {
-		return Boolean.TYPE;
+		return boolean.class;
 	}
 
 	public synchronized void dispose() {
