### Eclipse Workspace Patch 1.0
#P org.eclipse.core.databinding
Index: src/org/eclipse/core/databinding/observable/value/ValueChangeEvent.java
===================================================================
RCS file: /cvsroot/eclipse/org.eclipse.core.databinding/src/org/eclipse/core/databinding/observable/value/ValueChangeEvent.java,v
retrieving revision 1.4
diff -u -r1.4 ValueChangeEvent.java
--- src/org/eclipse/core/databinding/observable/value/ValueChangeEvent.java	16 Mar 2007 18:00:56 -0000	1.4
+++ src/org/eclipse/core/databinding/observable/value/ValueChangeEvent.java	20 Oct 2007 02:00:43 -0000
@@ -17,14 +17,14 @@
 /**
  * Value change event describing a change of an {@link IObservableValue}
  * object's current value.
- * 
+ *
  * @since 1.0
- * 
+ *
  */
 public class ValueChangeEvent extends ObservableEvent {
 
 	/**
-	 * 
+	 *
 	 */
 	private static final long serialVersionUID = 2305345286999701156L;
 
@@ -38,7 +38,7 @@
 
 	/**
 	 * Creates a new value change event.
-	 * 
+	 *
 	 * @param source
 	 *            the source observable value
 	 * @param diff
@@ -51,11 +51,11 @@
 
 	/**
 	 * Returns the observable value from which this event originated.
-	 * 
+	 *
 	 * @return returns the observable value from which this event originated
 	 */
 	public IObservableValue getObservableValue() {
-		return (IObservableValue) source;
+		return (IObservableValue) getSource();
 	}
 
 	protected void dispatch(IObservablesListener listener) {
