### Eclipse Workspace Patch 1.0
#P org.eclipse.equinox.common
Index: src/org/eclipse/core/runtime/CoreException.java
===================================================================
RCS file: /cvsroot/eclipse/org.eclipse.equinox.common/src/org/eclipse/core/runtime/CoreException.java,v
retrieving revision 1.5
diff -u -r1.5 CoreException.java
--- src/org/eclipse/core/runtime/CoreException.java	10 May 2006 18:23:18 -0000	1.5
+++ src/org/eclipse/core/runtime/CoreException.java	14 Nov 2007 22:25:19 -0000
@@ -11,7 +11,6 @@
 package org.eclipse.core.runtime;
 
 import java.io.PrintStream;
-import java.io.PrintWriter;
 
 /**
  * A checked exception representing a failure.
@@ -78,22 +77,22 @@
 			}
 		}
 	}
-
-	/**
-	 * Prints a stack trace out for the exception, and
-	 * any nested exception that it may have embedded in
-	 * its Status object.
-	 * 
-	 * @param output the stream to write to
-	 */
-	public void printStackTrace(PrintWriter output) {
-		synchronized (output) {
-			super.printStackTrace(output);
-			if (status.getException() != null) {
-				output.print(getClass().getName() + "[" + status.getCode() + "]: "); //$NON-NLS-1$ //$NON-NLS-2$
-				status.getException().printStackTrace(output);
-			}
-		}
-	}
+	//FIXME GWT
+	//	/**
+	//	 * Prints a stack trace out for the exception, and
+	//	 * any nested exception that it may have embedded in
+	//	 * its Status object.
+	//	 *
+	//	 * @param output the stream to write to
+	//	 */
+	//	public void printStackTrace(PrintWriter output) {
+	//		synchronized (output) {
+	//			super.printStackTrace(output);
+	//			if (status.getException() != null) {
+	//				output.print(getClass().getName() + "[" + status.getCode() + "]: "); //$NON-NLS-1$ //$NON-NLS-2$
+	//				status.getException().printStackTrace(output);
+	//			}
+	//		}
+	//	}
 
 }
